### ISSUE TITLE
A question about the tf.nn.dropout()

### ISSUE THREAD

**Issue by dongzwhitsz** [kratzert/finetune_alexnet_with_tensorflow#74]
the tf.nn.dropout() function make actions all the time. But the function is used only in train step, isn't it? It also make actions in validation and predict step.

**Comment by kratzert** [comment]
I think I don't understand your question. During training dropout drops connections with a probability of the given dropout rate and during inference, the dropout layer does not drop any connections but rescales the activation according to the dropout rate. This is the normal behavior of dropout. Or what do you mean?

**Comment by dongzwhitsz** [comment]
your code use the dropout in train, validation, test steps. I think you should add a condition to control the time of drop like following:
    if is_training: tf.nn.dropout()

**Comment by Caleb196x** [comment]
When you are using model to predict, just set the droprate as one. It’s a basic concept. 
On 03/01/2019 15:08, dongzwhitsz wrote:

your code use the dropout in train, validation, test steps. I think you should add a condition to control the time of drop like following:
if is_training: tf.nn.dropout()
